### ISSUE TITLE
the date-time format checker returns True for non-string types

### ISSUE THREAD

**Issue by filip-komarzyniec** [python-jsonschema/jsonschema#1520]
Quite self-explanatory. The minimal example:

```python
import jsonschema
datetime_checker = jsonschema.FormatChecker().checkers["date-time"][0]

datetime_checker(123)
>> True

datetime_checker({})
>> True
```

I understand this might be considered a bit hacky, as I'm accessing the checker directly instead of via `jsonschema.validate` method. 
However this is quite a shortcoming due to other, related issue: https://github.com/python-jsonschema/jsonschema/issues/1519

If you have a custom "date-time" checker then the `string` type is not enforced leading to unexpected errors.

**Comment by Julian** [comment]
That's correct behavior according to the JSON Schema specification.
